### PR TITLE
change: Upgrade Firestore and Storage Dependencies

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Upgraded Firestore and Storage dependencies to their latest versions. As a result the following updates were also needed.
  - Upgrade TypeScript to 4.x [breaking change!]
  - Update the `target` settings in `tsconfig`. Since we drop support for Node.js 12, adopted the [optimal settings for Node.js 14](https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-14).
 - Dropped support for Node.js 12 and updated the CIs to run on Node.js 14. Will also add Node.js 16 in a future PR.

RELEASE NOTE: Dropped support for Node.js 12. Developers should use Node.js 14 or higher when deploying the Admin SDK.

RELEASE NOTE: Upgraded TypeScript to v4.6.4

RELEASE NOTE: Upgraded the `@google-cloud/firestore` package to v5. This contains breaking changes. Refer to the Cloud Firestore [release notes](https://github.com/googleapis/nodejs-firestore/releases/tag/v5.0.0) for more details.

RELEASE NOTE: Upgraded the `@google-cloud/storage` package to v6. This contains breaking changes. Refer to the Cloud Storage [release notes](https://github.com/googleapis/nodejs-storage/releases/tag/v6.0.0) for more details.

### CODE REVIEW

**Review by egilmorez — APPROVED** [review]
Thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -87,8 +87,8 @@ information on using pull requests.
 
 ### Prerequisites
 
-1. Node.js 10.13.0 or higher.
-2. NPM 5 or higher (NPM 6 recommended).
+1. Node.js 14 or higher.
+2. NPM 6 or higher.
```
[the remark this answers is not in the document]
Comment (lahirumaramba) [reply CONTRIBUTING.md:91]: Acknowledged! `npm` is the correct term, let's use it here as well. Thanks!

**Review by dwyfrequency — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -87,8 +87,8 @@ information on using pull requests.
 
 ### Prerequisites
 
-1. Node.js 10.13.0 or higher.
-2. NPM 5 or higher (NPM 6 recommended).
+1. Node.js 14 or higher.
+2. `npm`` 6 or higher.
```
Comment (hsubox76) [inline CONTRIBUTING.md:91]: Is that an extra backtick?

File: tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "module": "commonjs",
-    "target": "es5",
+    "target": "es2020",
```
Comment (hsubox76) [inline tsconfig.json:4]: Why this change?

**Review by hsubox76 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (lahirumaramba) [reply CONTRIBUTING.md]: Typo! Thank you.
